lib/backends/sdc/utils.js

```diff
@@ -110,45 +110,46 @@ function getPublishingRules(opts, fwrules) {
  *
  * In 1.9.1:
  *
- *     filters={"label":["com.joyent.package=sample-2G"]}
+ *     filters={"label":["com.joyent.package=sample-2G", "foo=bar"]}
  *
  * In 1.10.0:
  *
- *     filters={"label":{"com.joyent.package=sample-2G":true}}
- *
- * This function takes the filters[label] value (as the filters parameter here)
- * and returns a normalized array of the packages we want to filter on.
+ *     filters={"label":{"com.joyent.package=sample-2G":true, "foo=bar":true}}
  *
+ * This function takes the filters object and returns a new normalized object
+ * with each object value being an array instance.
  */
-function getLabelFilters(opts, filters) {
+function getNormalizedFilters(opts, filters) {
     assert.object(opts, 'opts');
-    assert.number(opts.apiVersion, 'opts.apiVersion');
     assert.object(opts.log, 'opts.log');
     assert.object(filters, 'filters');
 
+    var i;
     var log = opts.log;
-    var val = filters;
-
-    if ((common.apiVersionCmp(opts.apiVersion, 1.22) >= 0)
-        && !Array.isArray(val)) {
-        // the version is >= 1.22 and we got a non array object, convert to
-        // array based on keys.
-        val = Object.keys(val);
-    }
-
-    // val should have either been an array, or we should have turned it into
-    // one. Any other value is illegal.
-    if (!Array.isArray(val)) {
-        log.warn({
-            filters: filters,
-            type: typeof (val),
-            val: JSON.stringify(val)
-        }, 'WARNING: filter value is not an array');
-
-        return new Error('invalid filter');
+    var name;
+    var names = Object.keys(filters);
+    var newFilters = {};
+    var val;
+
+    for (i = 0; i < names.length; i++) {
+        name = names[i];
+        val = filters[name];
+        if (Array.isArray(val)) {
+            newFilters[name] = val; // Nothing to convert.
+        } else if (typeof (val) === 'object') {
+            // Convert to an array of keys.
+            newFilters[name] = Object.keys(val);
+        } else {
+            log.warn({
+                filters: filters,
+                type: typeof (val),
+                val: JSON.stringify(val)
+            }, 'WARNING: filter %s is not an array or object', name);
+            return new Error('invalid filters');
```

Comment:
I'd drop the `log.warn` (I'm assuming we are going to raise the error for the response and then the audit log will show it).

Please have the Error message include details:  `return new Error(format('invalid filters: filter is not an array or object: %j', val));`